packages/@aws-cdk/aws-route53/lib/hosted-zone.ts

```diff
@@ -190,6 +191,14 @@ export interface PublicHostedZoneProps extends CommonHostedZoneProps {
    * @default false
    */
   readonly caaAmazon?: boolean;
+
+  /**
+   * The principle for which a assumed by role is created
+   * for record set updation
+   *
+   * @default - No delegation configuration
+   */
+  readonly crossAccountZoneDelegationPrinciple?: iam.IPrincipal;
```

Comment:
In addition to the typo change below, I think it would be worth considering building in multi-account trust here from the start. For example, the hosted zone is for 'example.com', and the user wants to delegate 'prod.example.com' to a prod account, and 'test.example.com' to a test account.

```suggestion
  readonly crossAccountZoneDelegationPrincipal?: iam.IPrincipal;
```

Reply:
I thought of that use case as well and figured that using `CompositePrincipal` here and inverting the control to consumer is the best way to implement it. Therefore I am accepting `IPrincipal`.
[CompositePrincipal](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-iam.CompositePrincipal.html)

Reply:
Fair enough, good call.